client/core/core.go

```diff
@@ -1152,6 +1152,28 @@ func (c *Core) SetWalletPassword(appPW []byte, assetID uint32, newPW []byte) err
 		return newError(missingWalletErr, "wallet for %s (%d) is not known", unbip(assetID), assetID)
 	}
 
+	// Connect if necessary.
+	wasConnected := wallet.connected()
+	if !wasConnected {
+		err := wallet.Connect(c.ctx)
+		if err != nil {
+			return newError(connectionErr, "SetWalletPassword connection error: %v", err)
+		}
+	}
+
+	// Check that the new password works.
+	wasUnlocked := wallet.unlocked()
+	err = wallet.Unlock(string(newPW), aYear)
+	if err != nil {
+		return newError(authErr, "Error unlocking wallet. Is the new password correct?: %v", err)
+	}
+
+	if !wasConnected {
+		wallet.Disconnect()
```

Comment:
Should this also `wallet.Lock()` before disconnecting?

Reply:
I believe this came up somewhere else recently. I didn't think it was in-scope for this PR, but I think that wallets should lock internally during disconnection.

Reply:
Fair enough, we'll address this in a follow up.